File: summer/WebContent/js/MS/utility/FrugalListStoreState.js
/**
 * FrugalListStoreState
 */
// This enum controls the growth to successively more complex storage models
define(["dojo/_base/declare"], function(declare){
    
    var FrugalListStoreState  = declare("FrugalListStoreState", Object, {});
    FrugalListStoreState.Success = 0;
    FrugalListStoreState.SingleItemList = 1; 
    FrugalListStoreState.ThreeItemList = 2;
    FrugalListStoreState.SixItemList = 3;
    FrugalListStoreState.Array = 4;
    
    return FrugalListStoreState;
});

File: org.summer.view.widget/src/org/summer/view/widget/utils/FrugalListStoreState.java
package org.summer.view.widget.utils;
// These classes implement a frugal storage model for lists of <t>.
    // Performance measurements show that many lists contain a single item.
    // Therefore this list is structured to prefer a list that contains a single
    // item, and does conservative growth to minimize the memory footprint.
 
    // This enum controls the growth to successively more complex storage models
 public   enum FrugalListStoreState
    {
        Success,
        SingleItemList,
        ThreeItemList,
        SixItemList,
        Array
    }
